File: Source/OregonTilth.Web/src/app/shared/models/forms/field-standard-times/field-standard-time-summary-dto.ts
import { FieldLaborActivityDto } from "../../generated/field-labor-activity-dto";
import { FieldUnitTypeDto } from "../../generated/field-unit-type-dto";
import { LaborTypeDto } from "../../generated/labor-type-dto";
import { MachineryDto } from "../../generated/machinery-dto";
import { TimeStudyDto } from "../../generated/time-study-dto";
import { TimeStudySummaryDto } from "../time-studies/time-studies-upsert-dto";

export class FieldStandardTimeSummaryDto {
    WorkbookID: number;
    FieldStandardTimeID: number;
    FieldLaborActivity: FieldLaborActivityDto;
    LaborType: LaborTypeDto;
    Machinery: MachineryDto;
    FieldUnitType: FieldUnitTypeDto;
    AverageMinutesPerFieldUnit: number;
    StandardTimePerUnit: number;
    TimeStudies: TimeStudySummaryDto[];
    FieldLaborActivityAndLaborTypeNameForDropdown: string;
    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

File: OregonTilth.Models/DataTransferObjects/Workbook/Forms/FieldStandardTime/FieldStandardTimeSummaryDto.cs


using System.Collections.Generic;

namespace OregonTilth.Models.DataTransferObjects
{
    public class FieldStandardTimeSummaryDto : IHasTimeStudies
    {
        public int? FieldStandardTimeID { get; set; }
        public int WorkbookID { get; set; }
        public FieldLaborActivitySummaryDto FieldLaborActivity { get; set; }
        public LaborTypeDto LaborType { get; set; }
        public MachinerySummaryDto Machinery { get; set; }
        public FieldUnitTypeDto FieldUnitType { get; set; }
        public IEnumerable<TimeStudySummaryDto> TimeStudies { get; set; }
        
        public string FieldLaborActivityAndLaborTypeNameForDropdown { get; set; }

        public decimal AverageMinutesPerFieldUnit { get; set; }
        public decimal? StandardTimePerUnit { get; set; }
    }
}